dm/syncer/causality.go

```diff
@@ -23,11 +23,27 @@ import (
 	"go.uber.org/zap"
 )
 
-// causality provides a simple mechanism to improve the concurrency of SQLs execution under the premise of ensuring correctness.
-// causality groups sqls that maybe contain causal relationships, and syncer executes them linearly.
-// if some conflicts exist in more than one groups, causality generate a conflict job and reset.
+// causality provides a simple mechanism to ensure correctness when we are running
+// DMLs concurrently.
+// As a table might have one or multiple keys, row changes might depend on other
+// row changes, together they form a dependency graph, only row changes without
+// dependency can run concurrently.
+// currently, row changes for a table from upstream are dispatched to DML workers
+// by their keys, including PK and UKs, to make sure row changes with same keys
```

Comment:
```suggestion
// by their keys, to make sure row changes with same keys
```

seems `key` is terminology in this context. The meaning should be explained when it first introduced, and maybe use *key* to make it italic